src/ert/_c_wrappers/enkf/enkf_main.py

```diff
@@ -487,18 +491,31 @@ def sample_prior(
                         init_file = init_file % realization_nr
                     grid_file = self.ensembleConfig().grid_file
                     assert grid_file is not None
-                    grid = xtgeo.grid_from_file(grid_file)
+
                     try:
+                        grid = xtgeo.grid_from_file(grid_file)
                         props = xtgeo.gridproperty_from_file(
-                            init_file, name=parameter, grid=grid
+                            init_file,
+                            name=parameter,
+                            grid=grid,
                         )
+                        data = props.get_npvalues1d(order="C", fill_value=-np.nan)
                     except PermissionError as err:
                         context_message = (
                             f"Failed to open init file for parameter {parameter!r}"
                         )
                         raise RuntimeError(context_message) from err
+                    except ValueError:
```

Comment:
In which cases do we expect to get a `ValueError`?
Is it when `grid_file` is for example made using `libecl` or something like that?